google/stackdriver_monitoring/stackdriver_handlers.py

```diff
@@ -98,78 +187,206 @@ def __init__(self, options, stackdriver):
 
   def __call__(self, request, path, params, fragment):
     project = params.get('project', self.__project)
+    type_map = self.__stackdriver.fetch_custom_descriptors(project)
 
     delete_method = (self.__stackdriver.service.projects()
                      .metricDescriptors().delete)
+    def delete_invocation(descriptor):
+      name = descriptor['name']
+      logging.info('batch DELETE %s', name)
+      return delete_method(name=name)
+    get_descriptor_name = lambda descriptor: descriptor['name']
 
-    type_map = self.__stackdriver.fetch_custom_descriptors(project)
+    handler = BatchProcessor(
+        project, self.__stackdriver,
+        type_map.values(), delete_invocation, get_descriptor_name)
+    handler.process()
 
-    all_names = [descriptor['name'] for descriptor in type_map.values()]
-    batch = self.__stackdriver.service.new_batch_http_request()
-    max_batch = 100
-    count = 0
+    response_code = 200 if handler.num_ok == len(type_map) else 500
```

Comment:
The builtin [http library](https://docs.python.org/2/library/httplib.html) allows you to replace these response codes with constants.